test.disabled/ejabberd_tests/tests/rest_helper.erl

```diff
@@ -147,6 +150,77 @@ fusco_request(Method, Path, Body, HeadersIn, Port, SSL) ->
     fusco_cp:stop(Client),
     Result.
 
+-spec get_port(role()) -> Port :: integer().
+get_port(Role) -> 
+    Listeners = apply_on_mim1(ejabberd_config, get_local_option, [listen]),
+    [{PortIpNet, ejabberd_cowboy, _Opts}] = lists:filter(fun(Config) -> is_roles_config(Config, Role) end, Listeners),
+    case PortIpNet of
+        {Port, _Host, _Net} -> Port;
+        {Port, _Host} -> Port;
+        Port -> Port
+    end.
+
+-spec get_ssl_status(role()) -> boolean().
+get_ssl_status(Role) ->
+    Listeners = apply_on_mim1(ejabberd_config, get_local_option, [listen]),
+    [{_PortIpNet, _Module, Opts}] = lists:filter(fun (Opts) -> is_roles_config(Opts, Role) end, Listeners),
+    lists:keymember(ssl, 1, Opts).
+
+% @doc Changes the control credentials for admin by restarting the listener
+% with new options.
+-spec change_admin_creds({User :: binary(), Password :: binary()}) -> 'ok' | 'error'.
+change_admin_creds(Creds) ->
+    stop_listener(admin),
```

Comment:
It's never called with role `client` so I think it should be named simply `stop_admin_listener()` and become a complimentary function to `start_admin_listener/1`.